src/drawing-tool/components/runtime.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useRef, useEffect } from "react";
+import DrawingTool from "drawing-tool";
+import 'drawing-tool/dist/drawing-tool.css';
+import { IAuthoredState, IInteractiveState } from "./app";
+import css from "./runtime.scss";
+import predefinedStampCollections from "./stamp-collections";
+import { renderHTML } from "../../shared/utilities/render-html";
+
+const kToolbarWidth = 40;
+const kToolbarHeight = 600;
+
+interface IProps {
+  authoredState: IAuthoredState;
+  interactiveState?: IInteractiveState | null;
+  setInteractiveState?: (updateFunc: (prevState: IInteractiveState | null) => IInteractiveState) => void;
+  report?: boolean;
+}
+
+interface StampCollections {
+  [collectionName: string]: string[];
+}
+
+interface DrawingToolOpts {
+  width: number;
+  height: number;
+  stamps?: StampCollections;
+}
+
+export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, setInteractiveState, report }) => {
+
+  const readOnly = !!(report || (authoredState.required && interactiveState?.submitted));
+
+  // equivalent to componentDidLoad
```

Comment:
This statement isn't 100% true. Sounds like nitpicking, but I've already seen problems related to this assumption. It's pretty common when we transition from Class-based components. If you merge in `master` branch, you should see quite a few errors/warning from ESLint (since TSLint has been replaced with ESLint with rules-of-hooks plugins in master). It could be fixed by adding dependencies to useEffect's second argument.